src/bun.js/bindings/webcore/JSDOMFormData.cpp

```diff
@@ -464,17 +465,23 @@ static inline JSC::EncodedJSValue jsDOMFormDataPrototypeFunction_set2Body(JSC::J
     auto filename = argument2.value().isUndefined() ? String() : convert<IDLUSVString>(*lexicalGlobalObject, argument2.value());
     RETURN_IF_EXCEPTION(throwScope, encodedJSValue());
 
-    RefPtr<Blob> blobValue = nullptr;
-    if (argument1.value().inherits<JSBlob>()) {
-        blobValue = Blob::create(argument1.value());
+    RefPtr<File> fileValue = nullptr;
+    auto filenameString = Bun::toString(filename);
+    if (argument1.value().inherits<JSFile>()) {
+        fileValue = File::create(argument1.value());
+    } else if (argument1.value().inherits<JSBlob>()) {
+        fileValue = File::fromJSBlob(argument1.value(), &filenameString);
+    } else {
+        throwTypeError(lexicalGlobalObject, throwScope, "Expected argument to be a Blob or File."_s);
+        RETURN_IF_EXCEPTION(throwScope, encodedJSValue());
     }
 
-    if (!blobValue) {
-        throwTypeError(lexicalGlobalObject, throwScope, "Expected argument to be a Blob."_s);
+    if (!fileValue) {
+        throwTypeError(lexicalGlobalObject, throwScope, "This should never be reached!"_s);
```

Comment:
This is the issue I'm currently stuck at: !fileValue causes fileValue to disappear, and I have no idea why. On the line before this, I can access properties of fileValue without a problem, for example get the name. But this causes it to be dropped for some reason.